snapshot/git/gitdb/schemes.go

```diff
@@ -0,0 +1,65 @@
+package gitdb
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/scootdev/scoot/snapshot"
+)
+
+// GitDB uses several different Schemes to identify, upload and download Values.
```

Comment:
(See also the design doc; should be in your email)

It's annoying in Unix filesystems that a file can be either a directory or... a file? Wait? What?

I'm proposing:
ScootDB stores Values.
A Value can be either a Snapshot or SnapshotWithHistory (for now; eventually we plan to store files, and who knows, maybe we just store a raw AST at some point).

So I think we need k + 1 terms: 1 for each of k types, plus for the baseclass/superclass.

WDYT?